tools/bin/mbedtls-archive-epic.py

```diff
@@ -0,0 +1,340 @@
+
+
+def github_querry(query, token=token):
+    """ Create an authenticated querry to github"""
+
+    request = requests.post('https://api.github.com/graphql',
+                            json={'query': query},
+                            headers={"Authorization": "token " + token})
+    if request.status_code == 200:
+        return request.json()
+    else:
+        raise Exception(
+            "Query failed with code {}. Query: {}".format(
+                request.status_code, query))
+
+
+def project_get_label_ids(project_id=project_past_epics):
+    """Use a gql querry to retrieve the node-ids for the column names of projectV2 """
+
+    labels = []
+    querry = gql_get_ids_querry.substitute(project_uid=project_id)
+    result = github_querry(querry)
+    # clean up the result
+    nodes = [n for n in result["data"]["node"]["fields"]["nodes"] if n]
+    return nodes.pop()["options"]
+
+
+####################### Github Cli Wrappers #########################
+def get_issues_prs_for_project(issues=True,
+                               is_closed=True,
+                               from_date=closed_after_date,
+                               count=max_entries,
+                               repo=projects_repo):
+    """ Retrieve a maximum n=count number issues or pull requests closed after a date """
+
+    cmd = ("gh {0} --repo {5} list -L {3} -s all --search \"is:{0} is:{1} "
+           "{2}:>{4}\"  --json 'number,projectItems'").format("issue" if issues else "pr",
+                                                                 "closed" if is_closed else "open",
+                                                                 "closed" if is_closed else "created",
+                                                                 count, from_date,
+                                                                 repo)
+    ret = do_shell_exec(cmd)
+    data = json.loads(ret.stdout)
+    data = [n for n in data if n["projectItems"]]
+
+    data = [{n["number"]: n["projectItems"]} for n in data]
+    return data
+
+def link_issues_pr_linked_to_epic(raw_data, epic_name, projectboard = ""):
+    """ Parse a raw data output and return a list of items linked to a selected epic."""
+
+    linked = []
+    for i in raw_data:
+        for k, v in i.items():
+            for e in v:
+                if epic_name and projectboard:
+                    if e["status"]["name"] == epic_name and e['title'] == projectboard:
+                        linked.append(k)
+                else:
+                    if e["status"]["name"] == epic_name:
+                        linked.append(k)
+    return linked
+
+def get_epic_items(column_name, status="both", projectboard=active_project, repo=projects_repo):
+    """ Return a structured data-set containing an epic name and all the open/closed issues in it."""
+
+    output = {"epic": column_name,
+              "issues": [], "prs": []}
+
+    if status not in ["open", "closed", "both"]:
```

Comment:
Should be an enum?

Or just not bother, since this function is always called with `status="both"`?